site/en/docs/extensions/mv3/getstarted/tut-focus-mode/index.md

```diff
@@ -0,0 +1,374 @@
+OFF), it will track each tab's state through the _extension badge_.
+
+{% Aside 'key-term' %}
+
+The [extension badge][action-badge] is a colored banner on top of the extension action (toolbar
+icon).
+
+{% endAside %}
+
+Create a file called `background.js` and add the following code to set the initial state to “OFF”:
+
+{% Label %}background.js:{% endLabel %}
+
+```js
+chrome.runtime.onInstalled.addListener(() => {
+  chrome.action.setBadgeText({
+    text: "OFF",
+  });
+});
+...
+```
+
+{% Details %}
+{% DetailsSummary %}
+💡 **What if I have a more complex state?**
+{% endDetailsSummary %}
+
+You can use the [Storage API][api-storage] to store state data.
+
+{% endDetails %}
+
+### Step 3: Enable the extension action {: #step-3 }
+
+The _extension action_ controls the extension’s toolbar icon. So whenever the user clicks on the
+extension action, it will either run some code (like in this example) or display a popup. Add the
+following code to declare the extension action in the manifest:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "action": {
+    "default_icon": {
+      "16": "images/icon-16.png",
+      "32": "images/icon-32.png",
+      "48": "images/icon-48.png",
+      "128": "images/icon-128.png"
+    }
+  },
+  ...
+}
+```
+
+#### Use the activeTab permission to protect user privacy {: #active-tab}
+
+The [`activeTab`][doc-active-tab] permission grants the extension _temporary_ ability to execute code
+on the currently active tab. It also allows access to [sensitive properties][active-tab-allows] of
+the current tab.
+
+This permission is enabled when the user **_invokes_** the extension. In this case, the user invokes the
+extension by clicking on the extension action.
+
+{% Details %}
+{% DetailsSummary %}
+💡 **What other user interactions enable the activeTab permission?**
+{% endDetailsSummary %}
+
+Other user gestures that enable the activeTab permission are the following:
+
+- Pressing a keyboard shortcut combination.
+- Selecting a context menu item.
+- Accepting a suggestion from the omnibox.
+- Opening an extension popup.
+
+{% endDetails %}
+
+The `activetab` permission allows users to _purposefully_ choose to run the extension on the
+currently focused tab; this way, it protects the user’s privacy. Another benefit is that it does not
+trigger a [permission warning][doc-perms-warning].
+
+To use the activetab permission, add it to the manifest's permission array:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "permissions": ["activeTab"],
+  ...
+}
+```
+
+### Step 4: Track the state of the current tab {: #step-4 }
+
+After the user clicks on the extension action, the extension will check if the URL matches a
+documentation page. Next, it will check the state of the current tab and set the next state. Add the
+following code to `background.js`:
+
+{% Label %}background.js:{% endLabel %}
+
+```js
```

Comment:
This entire code block is intended by two spaces. Is that intentional? If not, please remove the extra indentation.

Reply:
Also, in this case we may want to avoid using the lines added/removed syntax as it's a decent number of lines of code. Perhaps we could just say "add the following code to the end of your background.js file"?

Reply:
I will fix the spacing. I want to talk a bit more about the lines added/removed syntax. I'd like to get others' opinions before we change all the code samples

Reply:
Agreed offsite to leave the code samples with ellipsis for now